src/analysis/retail/monk/windwalker/modules/apl/AplCheck.tsx

```diff
@@ -0,0 +1,79 @@
+import aplCheck, { build, Condition } from 'parser/shared/metrics/apl';
+import {
+  and,
+  buffMissing,
+  buffPresent,
+  describe,
+  hasResource,
+  lastSpellCast,
+  not,
+  or,
+} from 'parser/shared/metrics/apl/conditions';
+import SPELLS from 'common/SPELLS';
+import { suggestion } from 'parser/core/Analyzer';
+import annotateTimeline from 'parser/shared/metrics/apl/annotate';
+import TALENTS from 'common/TALENTS/monk';
+import RESOURCE_TYPES from 'game/RESOURCE_TYPES';
+import { SpellLink } from 'interface';
+
+const serenityOr = (cond: Condition<any>) => or(cond, buffPresent(TALENTS.SERENITY_TALENT));
+
+const hasChi = (min: number) => hasResource(RESOURCE_TYPES.CHI, { atLeast: min });
+
+export const apl = build([
+  {
+    spell: TALENTS.FAELINE_STOMP_TALENT,
+    condition: buffMissing(TALENTS.FAELINE_HARMONY_TALENT),
+  },
+  {
+    spell: TALENTS.STRIKE_OF_THE_WINDLORD_TALENT,
+    condition: serenityOr(hasChi(2)),
+  },
+  TALENTS.WHIRLING_DRAGON_PUNCH_TALENT,
+  {
+    spell: TALENTS.RISING_SUN_KICK_TALENT,
+    condition: and(buffPresent(TALENTS.XUENS_BATTLEGEAR_TALENT), serenityOr(hasChi(2))),
+  },
+  {
+    spell: TALENTS.FISTS_OF_FURY_TALENT,
+    condition: serenityOr(hasChi(3)),
+  },
+  {
+    spell: TALENTS.RISING_SUN_KICK_TALENT,
+    condition: and(buffMissing(TALENTS.XUENS_BATTLEGEAR_TALENT), serenityOr(hasChi(2))),
+  },
```

Comment:
you can leave the `buffMissing` off here since it is implied by the `buffPresent` rule above it (if the buff is present, the higher rule will match and this rule won't be examined).

```suggestion
  {
    spell: TALENTS.RISING_SUN_KICK_TALENT,
    condition: serenityOr(hasChi(2)),
  },
```

Reply:
Fixed this, and also corrected the buff its tracking (should be `Pressure Point`)